src/scxml_sm_interface.cpp

```diff
@@ -136,6 +143,43 @@ ScxmlSMInterface::ScxmlSMInterface(const std::string& scxml_file)
   }
 }
 
+// use this to determine the next state in the state machine, given the name of the transition you'd like to query
+const QString ScxmlSMInterface::getNeighbor(const QString& state, const QString& search_text)
+{
+  QString next_state;
+
+  for (auto& pair : state_transition_map_.at(state))
+  {
+    if (pair.first == search_text)
+    {
+      return pair.second;
+    }
+  }
+  return next_state;
+}
+
+bool ScxmlSMInterface::eventExists(const QString& event, std::set<std::pair<QString, QString>> events)
+{
+  int i = 0;
+  const int j = events.size();
+  for (auto& pair : events)
+  {
+    if (pair.first == event)
+    {
+      return true;
+    }
+    else
+    {
+      i = i + 1;
+    }
+
+    if (i == j)
+    {
+      return false;
+    }
+  }
+}
```

Comment:
I think we can simplify this significantly:

```suggestion
  return std::any_of(events.begin(), events.end(), [&event](const std::pair<QString, QString>&){ return pair.first == event});
```

Reply:
Still relevant